Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -71,23 +72,18 @@ int compareEventLists(Kernel::Logger &logger, const EventList &el1, const EventL
     const auto &e1 = events1[i];
     const auto &e2 = events2[i];
 
-    bool diffpulse = false;
-    bool difftof = false;
-    bool diffweight = false;
-    if (std::abs(e1.pulseTime().totalNanoseconds() - e2.pulseTime().totalNanoseconds()) > tolPulse) {
-      diffpulse = true;
-      ++numdiffpulse;
-    }
-    if (std::abs(e1.tof() - e2.tof()) > tolTof) {
-      difftof = true;
-      ++numdifftof;
-    }
+    bool diffpulse =
```

Comment:
Would it be better to use [`EventList::equals()`](https://github.com/mantidproject/mantid/blob/main/Framework/DataObjects/src/EventList.cpp#L631) for this? The message wouldn't be as clear.

Reply:
I don't know enough about either this block of code or `EventList::equals()`, I just saw that this was inefficient and felt compelled to meddle with it.  I'll see if that can serve as a replacement

Reply:
Looking at the code, this method is called from another method `compareEventListsInDetail` (L994), which suggests the purpose of this function is precisely the more detailed error messages.  So I think this needs to stay, unfortunately.